adapter/internal/oasparser/model/open_api.go

```diff
@@ -105,14 +116,18 @@ func setResourcesOpenAPI(openAPI openapi3.Swagger) []Resource {
 						continue
 					}
 					endpoint := getHostandBasepathandPort(serverEntry.URL)
-					resource.productionUrls = append(resource.productionUrls, endpoint)
+					if endpoint != nil {
+						resource.productionUrls = append(resource.productionUrls, *endpoint)
+					} else {
```

Comment:
if the endpoint is not an error right? rather than not a nil

Reply:
Based on the changes made, the function `getHostandBasepathandPort` either returns a pointer to the endpoint or a nil. Hence, it will not return an error.